src/app-layer-htp.c

```diff
@@ -965,6 +965,10 @@ static AppLayerResult HTPHandleResponseData(Flow *f, void *htp_state,
                     if (bstr_cmp_c(h->value, "h2c") != 0) {
                         break;
                     }
+                    if (!AppLayerProtoDetectConfProtoDetectionEnabled("tcp", "http2")) {
```

Comment:
`AppLayerProtoDetectConfProtoDetectionEnabled` should not be used in the packet runtime.

Reply:
Using `AppLayerProtoDetectGetProtoName` then ?

Reply:
I would say some bool/flag set during registration, so we don't call routines that aren't optimized for the packet runtme

Reply:
Do you mean to create a new bool ?
Or is there one ?

(Did not see that before next PR.)